src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -321,20 +321,6 @@ private BinaryLogRecordKind Write(BuildCheckResultMessage e)
             return BinaryLogRecordKind.BuildCheckMessage;
         }
 
-        private BinaryLogRecordKind Write(BuildCheckResultWarning e)
```

Comment:
why remove this?

Reply:
It's handled as a part of https://github.com/dotnet/msbuild/blob/6dcd44ec025ef90ed5d6d832e7af49b6b2dc5b10/src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs#L482

We don't have any custom fields that need to be logged separately (yet)

Reply:
nit: add this to the description of the PR

Reply:
However then those events will be deserialized as ordinary warnings.
They are still distinguishable manually - when one knows what to expect. But we cannot e.g. choose to skip them automatically during replay.